authorization/invitation/service/invitation_service.go

```diff
@@ -421,16 +445,16 @@ func (s *invitationServiceImpl) Accept(ctx context.Context, token uuid.UUID) (st
 
 			err = s.Repositories().IdentityRoleRepository().Create(ctx, ir)
 			if err != nil {
-				return resourceID, err
+				return resourceID, redirectOnFailure, errs.Wrap(err, "failed to create identity role")
 			}
 		}
 
 		// Delete the invitation
-		s.Repositories().InvitationRepository().Delete(ctx, inv.InvitationID)
+		s.Repositories().InvitationRepository().Delete(ctx, invitationID)
 
-		// Return the resource ID
-		return inviteToResource.ResourceID, nil
+		// Return the resource ID and redirect path
+		return inviteToResource.ResourceID, redirectURL, nil
 	}
 
-	return "", nil
+	return resourceID, redirectURL, nil
```

Comment:
so what happens if the invitation was not for a membership or if the resourceID was nil ? AFAICT, the user is redirected to the success page, but nothing was done and the invite remains unchanged in the DB.